src/System.Text.Json/tests/Serialization/CustomConverterTests.List.cs

```diff
@@ -237,5 +238,52 @@ public static void ListConverterPolymorphic()
             jsonSerialized = JsonSerializer.Serialize(contraVariantList, options);
             Assert.Equal(json, jsonSerialized);
         }
+
+        private class MyModelWithListToStringElementsConverter
+        {
+            [JsonConverter(typeof(ListToStringElementsConverter))]
+            public List<string> Items { get; set; }
+        }
+
+        [Fact]
+        public static void CustomListWithJsonConverterAttribute()
+        {
+            const string json = @"{""Items"":[""hello"",1,true]}";
+
+            MyModelWithListToStringElementsConverter obj;
+
+            void Verify()
+            {
+                Assert.Equal("hello", obj.Items[0]);
+                Assert.Equal("1", obj.Items[1]);
+                Assert.Equal("True", obj.Items[2]);
+            }
+
+            obj = JsonSerializer.Deserialize<MyModelWithListToStringElementsConverter>(json);
```

Comment:
Can we first have a baseline test showing that an exception (JsonException, I believe) will be thrown if the attribute is not applied?

Reply:
Sure. Also added array and dictionary just in case (since they take different code paths prior to the method changed).